migrations/migration.go

```diff
@@ -273,38 +268,31 @@ func (m *StorageMigration) migrateNestedValue(
 				valueToSet,
 			)
 		}
+	}
 
-		// The dictionary itself does not have to be replaced
-		return
-	default:
-		// Assumption: all migrations only migrate non-container typed values.
-		for _, migration := range valueMigrations {
-			converted, err := m.migrate(migration, addressPath, value)
-
-			if err != nil {
-				if reporter != nil {
-					reporter.Error(addressPath, migration.Name(), err)
-				}
-				continue
+	for _, migration := range valueMigrations {
+		converted, err := migration.Migrate(addressPath, value, m.interpreter)
```

Comment:
`m.migrate` must be called here to gracefully handle runtime panics.

```suggestion
		converted, err := m.migrate(migration, addressPath, value)
```